From 09758ebb191da659563b4c33334295ce21ecae71 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Thu, 1 Apr 2021 07:49:46 +0200
Subject: [PATCH] lib/getrandom.c: fix build with uclibc < 1.0.35

Fix the following build failure with uclibc < 1.0.35 which is raised
since the addition of getrandom module in
https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=682cc4e678aceff32dea2a84b6e5147bdf5a20a7:

In file included from ./sys/random.h:40,
                 from getrandom.c:22:
/tmp/instance-0/output-1/host/x86_64-buildroot-linux-uclibc/sysroot/usr/include/sys/random.h:27:35: error: unknown type name 'size_t'
   27 | extern int getrandom(void *__buf, size_t count, unsigned int flags)
      |                                   ^~~~~~
/tmp/instance-0/output-1/host/x86_64-buildroot-linux-uclibc/sysroot/usr/include/sys/random.h:8:1: note: 'size_t' is defined in header '<stddef.h>'; did you forget to '#include <stddef.h>'?
    7 | #include <features.h>
  +++ |+#include <stddef.h>
    8 |

This patch is not upstreamable as it is only a workaround for
uclibc < 1.0.35, upstream uclibc has been patched with
https://github.com/wbx-github/uclibc-ng/commit/00972c02c2b6e0a95d5def4a71bdfb188e091782

Fixes:
 - http://autobuild.buildroot.org/results/30105094e39374ec6d4e47e2fb5e99696f7f7981

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: Not applicable (Buildroot specific)
[Julien: refreshed the patch to be applied with fuzz factor 0]
Signed-off-by: Julien Olivain <ju.o@free.fr>
---
 gl-libpoke/getrandom.c          | 1 +
 gl/getrandom.c                  | 1 +
 jitter/gnulib-local/getrandom.c | 1 +
 3 files changed, 3 insertions(+)

diff --git a/gl-libpoke/getrandom.c b/gl-libpoke/getrandom.c
index 19bb342..0251e0f 100644
--- a/gl-libpoke/getrandom.c
+++ b/gl-libpoke/getrandom.c
@@ -19,6 +19,7 @@
 
 #include <config.h>
 
+#include <stddef.h>
 #include <sys/random.h>
 
 #include <errno.h>
diff --git a/gl/getrandom.c b/gl/getrandom.c
index 19bb342..0251e0f 100644
--- a/gl/getrandom.c
+++ b/gl/getrandom.c
@@ -19,6 +19,7 @@
 
 #include <config.h>
 
+#include <stddef.h>
 #include <sys/random.h>
 
 #include <errno.h>
diff --git a/jitter/gnulib-local/getrandom.c b/jitter/gnulib-local/getrandom.c
index 19bb342..0251e0f 100644
--- a/jitter/gnulib-local/getrandom.c
+++ b/jitter/gnulib-local/getrandom.c
@@ -19,6 +19,7 @@
 
 #include <config.h>
 
+#include <stddef.h>
 #include <sys/random.h>
 
 #include <errno.h>
-- 
2.45.2

